astropy/table/table.py

```diff
@@ -1521,6 +1521,40 @@ def _repr_html_(self):
     def __repr__(self):
         return self._base_repr_(html=False, max_width=None)
 
+    def _repr_latex_ (self, latexdict={}):
+        """
+        Format Table object into the LaTeX format and print to the screen.
+
+        Parameters
+        ----------
+        latexdict : str or dict
+            Dictionary of LaTeX formatting options or either one of these three
+            strings corresponding to pre-defined options, ``AA``, ``doublelines``,
+            ``template``. Defaults to ``{}``.
+
+        Notes
+        -----
+        Uses ``latex.Latex.write()`` function from ``astropy.io.ascii`` to change
+        the table format to latex. The default LaTeX formatting options defined
+        in ``latex.Latex`` are assumed if no ``latexdict`` argument is passed.
+        See: https://docs.astropy.org/en/latest/api/astropy.io.ascii.Latex.html#astropy.io.ascii.Latex
+
+        A new key ``latexdict`` corresponding to ``Latex.latex`` dictionary of
+        ``astropy.io.ascii.latex`` object is added to ``table.Table.meta``.
+        It contains the LaTeX formatting options for the table used.
+        """
+        from astropy.io import ascii
+        from astropy.io.ascii import latex
+
+        if latexdict in ['AA', 'doublelines', 'template']:
+            latexdict = ascii.latex.latexdicts[latexdict]
+
+        lTable = latex.Latex(latexdict=latexdict)
+        lines = lTable.write(table=self)
+        self.meta['latexdict'] = lTable.latex
+        for line in lines:
+            print(line)
```

Comment:
The `_repr_*` functions should return, not print. The original use case was to have the latex representation automatically included in a latex file - for that it makes little sense to print it on the screen. 
Just join all your lines together and retur nthe string that has all of them.